service/history/historyEngine.go

```diff
@@ -824,16 +826,31 @@ func (e *historyEngineImpl) QueryWorkflow(
 		if err != nil {
 			return nil, err
 		}
-		result := state.queryResult
-		switch result.GetResultType() {
-		case workflow.QueryResultTypeAnswered:
-			return &h.QueryWorkflowResponse{
-				Response: &workflow.QueryWorkflowResponse{
-					QueryResult: result.GetAnswer(),
-				},
-			}, nil
-		case workflow.QueryResultTypeFailed:
-			return nil, &workflow.QueryFailedError{Message: result.GetErrorMessage()}
+		switch state.state {
```

Comment:
Move these into separate method.

Reply:
Here I also prefer to keep it as one chunk of code. The reason is all other methods in historyEngine do this. I think it's confusing to see all the methods look one way and then see this method spread into several functions.

Wenquan and I talked about a solution which I like: https://github.com/uber/cadence/issues/2748

Basically most of the methods in historyEngine should be their own file like what Wenquan did with decisionHandler. Then these files can be nicely factored.